### PR TITLE
Add command to prune unused secrets

### PR DESCRIPTION
**Pull request by taeold**
Each active secret version cost money. To help save cost on using Secret Manager, we add `functions:secrets:prune` command which:

1) Looks up all secret versions from secrets marked with label "firebase-managed". All secrets created using the Firebase CLI will have this label.

2) Look up all secret bindings for CF3 function instance.

3) Figure out which secret version isn't currently being used.

Since destroying a secret version is irrevocable and immediately breaking for clients that depend on it, we will always ask for a confirmation for the destroy operations (and not support -f flag).

Note that we now query `v1` of Secret Manager since `v1beta` does not offer filtering by labels.

### CODE REVIEW

**Review by inlined — APPROVED** [review]
Do customers need to call the prune command after setting new versions to avoid leaks?

File: src/commands/functions-secrets-prune.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+    // prompt to get them all deleted
+    logBullet(
+      `Found ${pruned.length} unused active secret versions:\n\t` +
+        pruned.map((sv) => `${sv.secret}@${sv.version}`).join("\n\t")
+    );
+
+    const confirm = await promptOnce(
+      {
+        name: "destroy",
```
[the remark this answers is not in the document]
Comment (taeold) [reply src/commands/functions-secrets-prune.ts:47]: Noted, I'll apply changes here and elsewhere.

File: src/functions/secrets.ts [hunk 2]
```diff
@@ -86,3 +96,59 @@ export async function ensureSecret(
...
+    let name = sev.secret;
+    if (name.includes("/")) {
+      const secret = parseSecretResourceName(name);
+      name = secret.name;
+    }
+
+    let version = sev.version;
+    if (version === "latest") {
+      // We need to figure out what "latest" resolves to.
+      const resolved = await getSecretVersion(projectId, name, version);
```
[the remark this answers is not in the document]
Comment (taeold) [reply src/functions/secrets.ts:144]: lol I understand, but this is really an edge case we are considering (e.g. users is using "latest" label on the secret in GCF, which is not possible in Firebase CLI)

**Comment by taeold** [comment]
> Do customers need to call the prune command after setting new versions to avoid leaks?

Yes, though we can improve the flow of --set to automatically deploy the affected functions.

**Pull request merged**